File: backend/ferbook/src/main/java/hr/fer/ppp/ferbook/api/model/Friendship.java
package hr.fer.ppp.ferbook.api.model;


import lombok.Getter;
import lombok.NoArgsConstructor;
import lombok.Setter;

import javax.persistence.*;
import java.sql.Timestamp;

@Entity
@Getter
@Setter
@NoArgsConstructor
public class Friendship {

    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private int ID;

    @ManyToOne
    private User user;

    @ManyToOne
    private User friend;

    private Timestamp acceptedInvitationAt;

    public Friendship(User user, User friend) {
        this.user = user;
        this.friend = friend;
    }
}

File: frontend/stores/models/friendship.js
import { User } from './user';

export class Friendship {
  constructor(data) {
    this.id = data.id;

    if (data.user instanceof User) {
      this.user = data.user;
    } else if (data.user) {
      this.user = new User(data.user);
    }

    if (data.friend instanceof User) {
      this.friend = data.friend;
    } else if (data.friend) {
      this.friend = new User(data.friend);
    }
  }

  serialize() {
    return {
      id: this.id,
      user: {
        id: this.user.id,
      },
      friend: this.friend && {
        id: this.friend.id,
      },
    }
  }
}